src/features/node-protocol.ts

```diff
@@ -1,18 +1,58 @@
+import { builtinModules } from 'node:module'
+import type { ResolvedOptions } from '..'
 import type { Plugin } from 'rolldown'
 
 /**
  * The `node:` protocol was added in Node.js v14.18.0.
  * @see https://nodejs.org/api/esm.html#node-imports
  */
-export function NodeProtocolPlugin(): Plugin {
+export function NodeProtocolPlugin(
+  options: Pick<ResolvedOptions, 'nodeProtocol' | 'removeNodeProtocol'>,
+): Plugin {
+  const nodeProtocolOption =
+    options.nodeProtocol ??
+    // `removeNodeProtocol: true` means stripping the `node:` protocol which equales to `nodeProtocol: 'strip'`
+    // `removeNodeProtocol: false` means keeping the `node:` protocol which equales to `nodeProtocol: false` (ignore it)
+    (options.removeNodeProtocol ? 'strip' : false)
```

Comment:
Move it to resolveOptions, and remove `removeNodeProtocol` from `ResolvedOptions`